gazebo_ros/include/gazebo_ros/executor.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAZEBO_ROS_EXECUTOR_HPP
+#define GAZEBO_ROS_EXECUTOR_HPP
+
+#include <rclcpp/rclcpp.hpp>
+#include <gazebo/common/common.hh>
+
+namespace gazebo_ros
+{
+
+/// Executor run in a seperate thread to handle events from all #gazebo_ros::Node instances
+/**
+ * \class Executor executor.hpp <gazebo_ros/executor.hpp>
+ */
+class Executor : public rclcpp::executors::MultiThreadedExecutor
+{
+public:
+  /// Create an instance and start the internal thread
+  Executor();
+
+  /// Stops the internal exeuctor and joins with the internal thread
+  virtual ~Executor();
+
+private:
+  /// Thread where the exeuctor spins until destruction
+  std::thread spin_thread_;
```

Comment:
Style nitpick: members should come after functions, see [guide](https://google.github.io/styleguide/cppguide.html#Declaration_Order)